### PR TITLE
Feature/sphinx tutorials

### PR DESCRIPTION
**Pull request by jcharkow**
Add Sphinx tutorials and also add __init__ for API documentation building. Good setup for doing relative imports

### CODE REVIEW

**Comment by jcharkow** [comment]
@singjc I think this is ready for review when you have a chance. My main concern is to ensure that  I did not break anything with the actual codebase. I had to remove some documentation in the codebase to get automatic sphinx documentation to work. I also added GenericChromatogramLoader and GenericSpectrumLoader because they are easier to describe in the docs.

File: docs/API.rst [hunk 1]
```diff
@@ -0,0 +1,138 @@
+API
+===
+
+Note: This section provides a comprehensive overview of all methods and classes avaliable in MassSeer. This content is generated automatically using sphinx autosummary and autodoc based on the python documentation.
```
Comment (singjc) [inline docs/API.rst]: Update MassSeer -> MassDash

File: docs/Installation.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+MassDash is available as a python package on the python package index, and can be installed by pip in a terminal or cmd prompt.
+
+.. code-block:: bash
+
+        pip install massdash
+
+Building from Source
+--------------------
+
+The source code is freely open and accessible on Github at https://github.com/Roestlab/massdash. The package can be installed by cloning and installing from source using pip.
```
Comment (singjc) [inline docs/Installation.rst]: Can you add the licence name(link) that it's open under

File: docs/Installation.rst [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+Install using pip
+
+.. code-block:: bash
+
+        pip install -e .
+
+Docker Image
+------------
+
+MassDash is available as a docker image on dockerhub `here <https://hub.docker.com/r/singjust/massdash>`_, ensure you have docker installed on your system, you can then pull the latest image of MassDash.
```
Comment (singjc) [inline docs/Installation.rst]: Currently I have a github workflow that automatically pushes to my dockerhub account, and to githubs container registry. Not sure which one we should use. The github one is directly linker under the packages in the github repo, so maybe that's the better choice to stick with?


https://github.com/Roestlab/massdash/pkgs/container/massdash

File: docs/conf.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+#
+import os
+import sys
+sys.path.insert(0, os.path.abspath('../massdash'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'MassDash'
+copyright = '2023, RoestLab'
```
Comment (singjc) [inline docs/conf.py]: Should we update the year? Can we automate the year?

```
from datetime import datetime
copyright = f'{datetime.now().year}, RoestLab'
```

File: docs/plottingGallery/Heatmap Per Transition.ipynb [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+     "ename": "ModuleNotFoundError",
+     "evalue": "No module named 'massseer'",
+     "output_type": "error",
+     "traceback": [
+      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
+      "\u001b[0;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
+      "Cell \u001b[0;32mIn[4], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mmassseer\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mplotting\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m PlotConfig, InteractiveTwoDimensionPlotter\n\u001b[1;32m      3\u001b[0m config \u001b[38;5;241m=\u001b[39m PlotConfig()\n\u001b[1;32m      4\u001b[0m config\u001b[38;5;241m.\u001b[39mcontext \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mjupyter\u001b[39m\u001b[38;5;124m'\u001b[39m\n",
+      "\u001b[0;31mModuleNotFoundError\u001b[0m: No module named 'massseer'"
+     ]
+    }
```
Comment (singjc) [inline docs/plottingGallery/Heatmap Per Transition.ipynb:208]: Is this old output stored in the notebook?

File: docs/plottingGallery/Heatmap Per Transition.ipynb [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+   ],
+   "source": [
+    "from massseer.plotting import PlotConfig, InteractiveTwoDimensionPlotter\n",
```
Comment (singjc) [inline docs/plottingGallery/Heatmap Per Transition.ipynb:211]: Change massseer -> massdash

File: docs/python_docs/Loading Data.rst [hunk 7]
```diff
@@ -0,0 +1,29 @@
+Loading Data
+================
+
+.. currentmodule:: massdash.loaders
+
+To load raw data into MassSeer, a loader object must be initiated. There are two types of loader classes
```
Comment (singjc) [inline docs/python_docs/Loading Data.rst:6]: MassSeer -> MassDash

File: docs/requirements.txt [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+psutil
+pyopenms
+scipy
+streamlit>=1.30.0
+streamlit-javascript
+tqdm
+upsetplot
+nbsphinx
+sphinx-copybutton
+sphinx-rtd-theme
```
Comment (singjc) [inline docs/requirements.txt:3]: Can you reorder these based on alphanetical order, unless there is an order based on which needs to be installed first.

File: massdash/loaders/GenericChromatogramLoader.py [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+            smooth (bool, optional): Whether to smooth the chromatogram. Defaults to True.
+            sgolay_polynomial_order (int, optional): Order of the polynomial to use for smoothing. Defaults to 3.
+            sgolay_frame_length (int, optional): Frame length to use for smoothing. Defaults to 11.
+            scale_intensity (bool, optional): Whether to scale the intensity of the chromatogram such that all chromatograms are individually normalized to 1. Defaults to False.
+
+        Returns: 
+            bokeh.plotting.figure.Figure: Bokeh figure object
+        '''
+
+        print("here")
```
Comment (singjc) [inline massdash/loaders/GenericChromatogramLoader.py]: Remove print statement

File: massdash/loaders/GenericChromatogramLoader.py [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+        ## TODO allow plotting of multiple chromatograms
+        if len(self.dataFiles_str) > 1:
+            raise NotImplementedError("Only one transition file is supported")
+ 
+        # load the transitionGroup for plotting
+        transitionGroup = list(self.loadTransitionGroups(seq, charge).values())[0]
+        print(transitionGroup)
```
Comment (singjc) [inline massdash/loaders/GenericChromatogramLoader.py:87]: remove print statement

File: massdash/loaders/GenericChromatogramLoader.py [hunk 11]
```diff
@@ -0,0 +1,92 @@
+import pandas as pd 
+
+from .GenericLoader import GenericLoader
+from ..structs import TransitionGroup
+from abc import ABC, abstractmethod
+from typing import List, Union, Literal
```
Comment (singjc) [inline massdash/loaders/GenericChromatogramLoader.py]: Can you reorder these imports into

i.e.

```
from abc import ABC, abstractmethod
from typing import List, Union, Literal
import pandas as pd

# Loader
from .GenericLoader import GenericLoader
# Structs
from ..structs import TransitionGroup
```

File: massdash/loaders/GenericSpectrumLoader.py [hunk 12]
```diff
@@ -0,0 +1,116 @@
+'''
+This is an abstract class for loading spectra from a file.
+'''
+
+import pandas as pd
+
+from .GenericLoader import GenericLoader
+from abc import ABC, abstractmethod
+from ..structs import TransitionGroup, TargetedDIAConfig, FeatureMap
+from typing import List, Union, Literal, Optional
```
Comment (singjc) [inline massdash/loaders/GenericSpectrumLoader.py]: Reorder imports

File: massdash/loaders/__init__.py [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+from .GenericLoader import GenericLoader
+from .GenericChromatogramLoader import GenericChromatogramLoader
+from .GenericSpectrumLoader import GenericSpectrumLoader
+
+__all__ = [ "MzMLDataLoader", 
+            "SqMassLoader",
+            "SpectralLibraryLoader",
+            "GenericChromatogramLoader",
+            "GenericLoader",
+            "GenericSpectrumLoader"]
```
Comment (singjc) [inline massdash/loaders/__init__.py]: Can you order these based on alphabet?

File: massdash/plotting/GenericPlotter.py [hunk 14]
```diff
@@ -16,27 +16,36 @@ class PlotConfig:
...
+        normalization_dict (dict): A dictionary containing the parameters for normalizing the data (2D heatmap only)
     """
     def __init__(self):
+        """
+        A class representing the configuration for a plot.
+
+        Attributes:
+           
+
+
```
Comment (singjc) [inline massdash/plotting/GenericPlotter.py]: This can be removed?

File: massdash/structs/__init__.py [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+
+__all__ = [ "TransitionGroupFeature", 
+            "TransitionGroup", 
+            "TransitionFeature", 
+            "Chromatogram", 
+            "FeatureMap", 
+            "Mobilogram", 
+            "Spectrum",
+            "Data1D",
+             "TargetedDIAConfig" ]
```
Comment (singjc) [inline massdash/structs/__init__.py]: There seems to be an extra space in front of "TargetedDIAConfig"

**Review by singjc — CHANGES_REQUESTED** [review]
Looks good, great job! Just a few minor comments / changes.

Can you also test the GUI to make sure nothing is broken there, and it still works as expected with these changes.

File: docs/Installation.rst [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+        pip install -e .
+
+Docker Image
+------------
+
+MassDash is available as a docker image on dockerhub `here <https://github.com/Roestlab/massdash/pkgs/container/massdash>`_, ensure you have docker installed on your system, you can then pull the latest image of MassDash.
+
+.. code-block:: bash
+
+        docker pull singjust/massdash:latest
```
Comment (singjc) [inline docs/Installation.rst]: Can you update the pull statement as well

`docker pull ghcr.io/roestlab/massdash:latest`

File: docs/requirements.txt
Comment (singjc) [inline docs/requirements.txt:1]: is it possible to have a separate requirements.txt file that is strictly for documentation? Kind of like requirements-dev.txt?

**Review by singjc — APPROVED** [review]
Minor comments/questions

File: docs/requirements.txt
Comment (jcharkow) [reply docs/requirements.txt:1]: the requirements.txt for documentation is in the doc folder so it is separate

**Pull request merged**